src/python/pants/backend/docker/goals/package_image.py

```diff
@@ -154,11 +160,37 @@ def image_refs(
         )
 
 
-def get_build_options(target: Target) -> Iterator[str]:
+def get_build_options(
+    build_target_stage: str | None,
+    context: DockerBuildContext,
+    field_set: DockerFieldSet,
+    target: Target,
+) -> Iterator[str]:
+    # Build options from target fields inheriting from DockerBuildOptionFieldMixin
     for field_type in target.field_types:
         if issubclass(field_type, DockerBuildOptionFieldMixin):
             yield from target[field_type].options()
 
+    # Target stage
+    target_stage = None
+    if build_target_stage in context.stages:
+        target_stage = build_target_stage
+    elif field_set.target_stage.value:
+        target_stage = field_set.target_stage.value
+
+    if target_stage:
+        if target_stage not in context.stages:
+            raise DockerBuildTargetStageError(
```

Comment:
I think it will be more clear if this is moved to the `elif` above, as it can only happen in that instance. And then you can also explicitly reference the field's name in the error

Reply:
Good move. However, I fail to find a reasonable phrasing incorporating the field alias in the error message. I think it is pretty clear already.
```
docker_image(name="ex", target_stage="dummy", ...)
```

> "Attempt to build stage "dummy" for `docker_image` src/docker:ex, but there is no such stage in `src/docker/Dockerfile`. Available stages: base, dev, prod.

Reply:
`InvalidFieldException` has a template in its docstring:

> The {InvalidFieldException.alias!r} field in target {address} was set to "dummy", but there is no such stage in src/docker/Dockerfile. Available stages: base, dev, prod.